src/utils/index.ts

```diff
@@ -0,0 +1,35 @@
+const generateRedirectUrl = (
+  redirectTo: string,
+  queryParameters: { [key: string]: string },
+  hashTag?: string
+): string => {
+  let finalRedirectTo = redirectTo;
+
+  // add query paramters
+  // add first ? or & depending on of there are already query parameters
```

Comment:
why are we using both `#` and `?`? Wouldn't it be easier to put everything after `#` next to `refreshToken` and `type`?

Reply:
There is generally better support to get (and set) query parameters (`?` and `&`) than getting info from a hashtag (`#`).

The only reason we put the refresh token after a hashtag (instead of a regular query parameter) is because of security concerns. Data after a hashtag can not be read by the server, won't be saved in server logs, etc.